From d90603f5d7725accac3a56abc0c1a0ed84a6814b Mon Sep 17 00:00:00 2001
From: "Daniel A. Wozniak" <dwozniak@saltstack.com>
Date: Fri, 31 Jan 2020 00:12:45 +0000
Subject: [PATCH] singledispatch only needed for < 3.4

backports-abc only for py2

Allow pycrypto to be included in parsed requirements

Replace pycryptodome with pycrypto

Due to issues found during package testing for the several supported
linux distributions, we're switching back our crypto depencency to
pycrypto.

Security concerned users are still advised to install the latest m2crypto
or pycryptodome library after pip installing salt since salt is prepared
to work with whatever is available.
---
 .pre-commit-config.yaml               | 13 -------------
 pkg/osx/req.txt                       |  4 ++--
 pkg/windows/req.txt                   |  4 ++--
 requirements/base.txt                 |  4 ++--
 requirements/crypto.txt               |  2 +-
 requirements/static/linux.in          |  3 ---
 requirements/static/py2.7/darwin.txt  |  6 +++---
 requirements/static/py2.7/linux.txt   |  7 ++++---
 requirements/static/py2.7/windows.txt |  4 ++--
 requirements/static/py3.4/linux.txt   |  7 +++----
 requirements/static/py3.5/darwin.txt  |  6 ++----
 requirements/static/py3.5/linux.txt   |  7 +++----
 requirements/static/py3.5/windows.txt |  4 +---
 requirements/static/py3.6/darwin.txt  |  6 ++----
 requirements/static/py3.6/linux.txt   |  7 +++----
 requirements/static/py3.6/windows.txt |  4 +---
 requirements/static/py3.7/darwin.txt  |  6 ++----
 requirements/static/py3.7/linux.txt   |  8 ++++----
 requirements/static/py3.7/windows.txt |  4 +---
 setup.py                              |  2 +-
 20 files changed, 39 insertions(+), 69 deletions(-)

diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index dc6942ab37c5b02fbe5e0a365153a8853d411178..25cf9b07077c666696710da0820c8ccccd376d0b 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -15,7 +15,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-darwin-py2.7-zmq-requirements
@@ -30,7 +29,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
           - --passthrough-line-from-input=^pyobjc(.*)$
 
       - id: pip-tools-compile
@@ -46,7 +44,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-cloud-py2.7-requirements
@@ -99,7 +96,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-cloud-py3.4-requirements
@@ -132,7 +128,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-darwin-py3.5-zmq-requirements
@@ -147,7 +142,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
           - --passthrough-line-from-input=^pyobjc(.*)$
 
       - id: pip-tools-compile
@@ -163,7 +157,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-cloud-py3.5-requirements
@@ -234,7 +227,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-darwin-py3.6-zmq-requirements
@@ -249,7 +241,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
           - --passthrough-line-from-input=^pyobjc(.*)$
 
       - id: pip-tools-compile
@@ -265,7 +256,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-cloud-py3.6-requirements
@@ -336,7 +326,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-darwin-py3.7-zmq-requirements
@@ -351,7 +340,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
           - --passthrough-line-from-input=^pyobjc(.*)$
 
       - id: pip-tools-compile
@@ -367,7 +355,6 @@ repos:
           - --include=requirements/base.txt
           - --include=requirements/zeromq.txt
           - --include=requirements/pytest.txt
-          - --remove-line=^pycrypto==(.*)$
 
       - id: pip-tools-compile
         alias: compile-cloud-py3.7-requirements
diff --git a/pkg/osx/req.txt b/pkg/osx/req.txt
index 1f2a612b1c524a87d80f4c17fbed9e55f69276b6..b51ee6917f43ac05f85ac5ed26ed70773f132bd3 100644
--- a/pkg/osx/req.txt
+++ b/pkg/osx/req.txt
@@ -1,6 +1,6 @@
 apache-libcloud==2.4.0
 backports.ssl_match_hostname==3.7.0.1
-backports_abc==0.5
+backports-abc==0.5; python_version < '3.0'
 certifi
 cffi==1.12.2
 CherryPy==17.4.1
@@ -28,7 +28,7 @@ pyyaml==5.1.2
 pyzmq==18.0.1
 requests==2.21.0
 setproctitle
-singledispatch==3.4.0.3
+singledispatch==3.4.0.3; python_version < '3.4'
 smmap==0.9.0
 timelib==0.2.4
 vultr==1.0.1
diff --git a/pkg/windows/req.txt b/pkg/windows/req.txt
index 3a428417b3b397227af3086ff41c86cb6306cde3..e0f2f00ee7e308ebe90353f8b2d86b42db90fa36 100644
--- a/pkg/windows/req.txt
+++ b/pkg/windows/req.txt
@@ -1,5 +1,5 @@
 -r req_win.txt
-backports-abc==0.5
+backports-abc==0.5; python_version < '3.0'
 backports.ssl-match-hostname==3.7.0.1
 certifi
 cffi==1.12.2
@@ -33,7 +33,7 @@ pyyaml==5.1.2
 pyzmq==18.0.1
 requests==2.21.0
 setproctitle
-singledispatch==3.4.0.3
+singledispatch==3.4.0.3; python_version < '3.4'
 smmap==0.9.0
 timelib==0.2.4
 watchdog==0.9.0
diff --git a/requirements/base.txt b/requirements/base.txt
index e8fac5b525545b4e981182cfd961dbb5e5af8ab5..922aec4c754178fd5c317ed636a0ebe487fcb25d 100644
--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -4,7 +4,7 @@ PyYAML
 MarkupSafe
 requests>=1.0.0
 # Requirements for Tornado 4.5.3 (vendored as salt.ext.tornado)
-backports-abc==0.5
-singledispatch==3.4.0.3
+backports-abc==0.5; python_version < '3.0'
+singledispatch==3.4.0.3; python_version < '3.4'
 # Required by Tornado to handle threads stuff.
 futures>=2.0; python_version < '3.0'
diff --git a/requirements/crypto.txt b/requirements/crypto.txt
index c98b7f77656dcf73b786dbbb7dd6144ef9320782..9a62bb31f808cc1d8c91b8372a02bf785fa6bf91 100644
--- a/requirements/crypto.txt
+++ b/requirements/crypto.txt
@@ -1,2 +1,2 @@
-pycryptodome; sys.platform != 'win32'
+pycrypto>=2.6.1; sys.platform not in 'win32,darwin'
 pycryptodomex; sys.platform == 'win32'
diff --git a/requirements/static/linux.in b/requirements/static/linux.in
index f818655e24a4b547b5a68f71194dbf689477d7e1..c20b006e2bf7820f461818ebc284b4efa2004ea0 100644
--- a/requirements/static/linux.in
+++ b/requirements/static/linux.in
@@ -23,9 +23,6 @@ more-itertools==5.0.0
 moto
 paramiko>=2.1.6
 psutil
-# Let's install pycryptodome instead of pycrypto because of pycrypto's outstanding security issues
-# PyCrypto, if pulled, will be removed from the generated static requirements
-pycryptodome
 pygit2
 pyinotify
 pyopenssl
diff --git a/requirements/static/py2.7/darwin.txt b/requirements/static/py2.7/darwin.txt
index 9be7b096ba19ca479d28bec8d1ad88d5cd713c15..07bf63a78a675a394d962b785e19446b093cc13d 100644
--- a/requirements/static/py2.7/darwin.txt
+++ b/requirements/static/py2.7/darwin.txt
@@ -10,7 +10,7 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
+backports-abc==0.5 ; python_version < "3.0"
 backports.functools-lru-cache==1.5  # via cheroot, jaraco.functools
 backports.ssl-match-hostname==3.7.0.1
 backports.tempfile==1.0   # via moto
@@ -87,7 +87,7 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5
 pycparser==2.19
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycryptodome==3.8.1
 pynacl==1.3.0             # via paramiko
 pyopenssl==19.0.0
 pyparsing==2.4.5          # via packaging
@@ -115,7 +115,7 @@ scandir==1.10.0           # via pathlib2
 scp==0.13.2               # via junos-eznc
 selectors2==2.0.1         # via ncclient
 setproctitle==1.1.10
-singledispatch==3.4.0.3
+singledispatch==3.4.0.3 ; python_version < "3.4"
 six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
diff --git a/requirements/static/py2.7/linux.txt b/requirements/static/py2.7/linux.txt
index 6b995bd27fbe3fa8c4a99f0b9725974081b2cf8e..b6bab5e0f61b347e0f15dcc9539c3b70e4aa42c5 100644
--- a/requirements/static/py2.7/linux.txt
+++ b/requirements/static/py2.7/linux.txt
@@ -10,7 +10,7 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
+backports-abc==0.5 ; python_version < "3.0"
 backports.functools-lru-cache==1.5  # via cheroot, jaraco.functools
 backports.ssl-match-hostname==3.7.0.1  # via docker, websocket-client
 backports.tempfile==1.0   # via moto
@@ -85,7 +85,8 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5             # via paramiko, pyasn1-modules, rsa
 pycparser==2.19           # via cffi
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycrypto==2.6.1 ; sys_platform not in "win32,darwin"
+pycryptodome==3.8.1       # via python-jose
 pygit2==0.28.2
 pyinotify==0.9.6
 pynacl==1.3.0             # via paramiko
@@ -116,7 +117,7 @@ scp==0.13.2               # via junos-eznc
 selectors2==2.0.1         # via ncclient
 setproctitle==1.1.10
 setuptools-scm==3.2.0
-singledispatch==3.4.0.3
+singledispatch==3.4.0.3 ; python_version < "3.4"
 six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 strict-rfc3339==0.7
diff --git a/requirements/static/py2.7/windows.txt b/requirements/static/py2.7/windows.txt
index 52390fdbc2d95b1ca590e526f7c30faede395ad4..340ca703070ec5cee6a2e58e1f6e681d735e86d2 100644
--- a/requirements/static/py2.7/windows.txt
+++ b/requirements/static/py2.7/windows.txt
@@ -9,7 +9,7 @@ asn1crypto==0.24.0        # via cryptography
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
+backports-abc==0.5 ; python_version < "3.0"
 backports.functools-lru-cache==1.5  # via cheroot, jaraco.functools
 backports.ssl-match-hostname==3.7.0.1
 backports.tempfile==1.0   # via moto
@@ -111,7 +111,7 @@ salttesting==2017.6.1
 scandir==1.10.0           # via pathlib2
 sed==0.3.1
 setproctitle==1.1.10
-singledispatch==3.4.0.3
+singledispatch==3.4.0.3 ; python_version < "3.4"
 six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pathlib2, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
diff --git a/requirements/static/py3.4/linux.txt b/requirements/static/py3.4/linux.txt
index e6f73d06abe194ddc2e4c3c0b6c809a5aa37553b..4bc5ed8cc94126c23636f7038733139bc3e6204f 100644
--- a/requirements/static/py3.4/linux.txt
+++ b/requirements/static/py3.4/linux.txt
@@ -10,7 +10,6 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl-match-hostname==3.7.0.1  # via docker, websocket-client
 bcrypt==3.1.6             # via paramiko
@@ -76,7 +75,8 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5             # via paramiko, pyasn1-modules, rsa
 pycparser==2.19           # via cffi
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycrypto==2.6.1 ; sys_platform not in "win32,darwin"
+pycryptodome==3.8.1       # via python-jose
 pygit2==0.28.2
 pyinotify==0.9.6
 pynacl==1.3.0             # via paramiko
@@ -107,8 +107,7 @@ scp==0.13.2               # via junos-eznc
 selectors2==2.0.1         # via ncclient
 setproctitle==1.1.10
 setuptools-scm==3.2.0
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 strict-rfc3339==0.7
 tempora==1.14.1           # via portend
diff --git a/requirements/static/py3.5/darwin.txt b/requirements/static/py3.5/darwin.txt
index 459635efcd28446eb497bc065f4579b02b3f8268..ac3eb437da29fef31490c2c8a524916a022d7a44 100644
--- a/requirements/static/py3.5/darwin.txt
+++ b/requirements/static/py3.5/darwin.txt
@@ -10,7 +10,6 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl_match_hostname==3.7.0.1
 bcrypt==3.1.6             # via paramiko
@@ -79,7 +78,7 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5
 pycparser==2.19
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycryptodome==3.8.1
 pynacl==1.3.0             # via paramiko
 pyopenssl==19.0.0
 pyparsing==2.4.5          # via packaging
@@ -105,8 +104,7 @@ s3transfer==0.2.0         # via boto3
 salttesting==2017.6.1
 scp==0.13.2               # via junos-eznc
 setproctitle==1.1.10
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
 strict-rfc3339==0.7
diff --git a/requirements/static/py3.5/linux.txt b/requirements/static/py3.5/linux.txt
index 5615898767d39fe28d768abf330d29967cba16ff..c3611cfbcc7e154e6251d98d26a0dfc273a74fb3 100644
--- a/requirements/static/py3.5/linux.txt
+++ b/requirements/static/py3.5/linux.txt
@@ -10,7 +10,6 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl-match-hostname==3.7.0.1  # via websocket-client
 bcrypt==3.1.6             # via paramiko
@@ -76,7 +75,8 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5             # via paramiko, pyasn1-modules, rsa
 pycparser==2.19           # via cffi
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycrypto==2.6.1 ; sys_platform not in "win32,darwin"
+pycryptodome==3.8.1       # via python-jose
 pygit2==0.28.2
 pyinotify==0.9.6
 pynacl==1.3.0             # via paramiko
@@ -105,8 +105,7 @@ salttesting==2017.6.1
 scp==0.13.2               # via junos-eznc
 setproctitle==1.1.10
 setuptools-scm==3.2.0
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pathlib2, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 strict-rfc3339==0.7
 tempora==1.14.1           # via portend
diff --git a/requirements/static/py3.5/windows.txt b/requirements/static/py3.5/windows.txt
index b7679a49f0e21225d340f463d5f85cc8656b4420..ff3940c2a86ffebe07cc9b042dbd3e68e3e50b4a 100644
--- a/requirements/static/py3.5/windows.txt
+++ b/requirements/static/py3.5/windows.txt
@@ -9,7 +9,6 @@ asn1crypto==0.24.0        # via cryptography
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl-match-hostname==3.7.0.1
 boto3==1.9.132
@@ -102,8 +101,7 @@ s3transfer==0.2.0         # via boto3
 salttesting==2017.6.1
 sed==0.3.1
 setproctitle==1.1.10
-singledispatch==3.4.0.3
-six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pathlib2, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, websocket-client
+six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pathlib2, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
 strict-rfc3339==0.7
diff --git a/requirements/static/py3.6/darwin.txt b/requirements/static/py3.6/darwin.txt
index 6f8ab1e1cf8e5f2cc61d2c36d92b4ae5ca6be6e3..37d295a54f8e0ea9a78ca585e988494e0badc302 100644
--- a/requirements/static/py3.6/darwin.txt
+++ b/requirements/static/py3.6/darwin.txt
@@ -12,7 +12,6 @@ attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl_match_hostname==3.7.0.1
-backports_abc==0.5
 bcrypt==3.1.6             # via paramiko
 boto3==1.9.132
 boto==2.49.0
@@ -78,7 +77,7 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5
 pycparser==2.19
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycryptodome==3.8.1
 pynacl==1.3.0             # via paramiko
 pyopenssl==19.0.0
 pyparsing==2.4.5          # via packaging
@@ -104,8 +103,7 @@ s3transfer==0.2.0         # via boto3
 salttesting==2017.6.1
 scp==0.13.2               # via junos-eznc
 setproctitle==1.1.10
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
 strict-rfc3339==0.7
diff --git a/requirements/static/py3.6/linux.txt b/requirements/static/py3.6/linux.txt
index 2c237f0a8f5013a65adac7b8889ea87d7c8260b0..247bdc6e2bee3f6859663a1e787592caba4a6775 100644
--- a/requirements/static/py3.6/linux.txt
+++ b/requirements/static/py3.6/linux.txt
@@ -10,7 +10,6 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl-match-hostname==3.7.0.1  # via websocket-client
 bcrypt==3.1.6             # via paramiko
@@ -75,7 +74,8 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5             # via paramiko, pyasn1-modules, rsa
 pycparser==2.19           # via cffi
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycrypto==2.6.1 ; sys_platform not in "win32,darwin"
+pycryptodome==3.8.1       # via python-jose
 pygit2==0.28.2
 pyinotify==0.9.6
 pynacl==1.3.0             # via paramiko
@@ -104,8 +104,7 @@ salttesting==2017.6.1
 scp==0.13.2               # via junos-eznc
 setproctitle==1.1.10
 setuptools-scm==3.2.0
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 strict-rfc3339==0.7
 tempora==1.14.1           # via portend
diff --git a/requirements/static/py3.6/windows.txt b/requirements/static/py3.6/windows.txt
index 616a636e73d693fea3bb697c4bdaac9087b4127d..73638629c2e72d5a8568a82e629f9d23a5487dd4 100644
--- a/requirements/static/py3.6/windows.txt
+++ b/requirements/static/py3.6/windows.txt
@@ -9,7 +9,6 @@ asn1crypto==0.24.0        # via cryptography
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl-match-hostname==3.7.0.1
 boto3==1.9.132
@@ -101,8 +100,7 @@ s3transfer==0.2.0         # via boto3
 salttesting==2017.6.1
 sed==0.3.1
 setproctitle==1.1.10
-singledispatch==3.4.0.3
-six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, websocket-client
+six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
 strict-rfc3339==0.7
diff --git a/requirements/static/py3.7/darwin.txt b/requirements/static/py3.7/darwin.txt
index 5a73e5ae1441a4b8c299b23c6c8a71105cbdc594..5c41584fad487be064238ea418b38b41998e3b55 100644
--- a/requirements/static/py3.7/darwin.txt
+++ b/requirements/static/py3.7/darwin.txt
@@ -10,7 +10,6 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl_match_hostname==3.7.0.1
 bcrypt==3.1.6             # via paramiko
@@ -78,7 +77,7 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5
 pycparser==2.19
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycryptodome==3.8.1
 pynacl==1.3.0             # via paramiko
 pyopenssl==19.0.0
 pyparsing==2.4.5          # via packaging
@@ -104,8 +103,7 @@ s3transfer==0.2.0         # via boto3
 salttesting==2017.6.1
 scp==0.13.2               # via junos-eznc
 setproctitle==1.1.10
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kubernetes, mock, more-itertools, moto, ncclient, packaging, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
 strict-rfc3339==0.7
diff --git a/requirements/static/py3.7/linux.txt b/requirements/static/py3.7/linux.txt
index fdbc02a249d584be42d76ca2d12b73144b20a6ae..17e9bc785ae5e34ecefc4f6ad7c8b2b34f7651d5 100644
--- a/requirements/static/py3.7/linux.txt
+++ b/requirements/static/py3.7/linux.txt
@@ -10,8 +10,8 @@ asn1crypto==1.3.0         # via certvalidator, cryptography, oscrypto
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
+backports.ssl-match-hostname==3.7.0.1  # via websocket-client
 bcrypt==3.1.6             # via paramiko
 boto3==1.9.132
 boto==2.49.0
@@ -74,7 +74,8 @@ pyaml==19.4.1             # via moto
 pyasn1-modules==0.2.4     # via google-auth
 pyasn1==0.4.5             # via paramiko, pyasn1-modules, rsa
 pycparser==2.19           # via cffi
-pycryptodome==3.8.1 ; sys_platform != "win32"
+pycrypto==2.6.1 ; sys_platform not in "win32,darwin"
+pycryptodome==3.8.1       # via python-jose
 pygit2==0.28.2
 pyinotify==0.9.6
 pynacl==1.3.0             # via paramiko
@@ -103,8 +104,7 @@ salttesting==2017.6.1
 scp==0.13.2               # via junos-eznc
 setproctitle==1.1.10
 setuptools-scm==3.2.0
-singledispatch==3.4.0.3
-six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, vcert, websocket-client
+six==1.12.0               # via bcrypt, cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, junos-eznc, kazoo, kubernetes, mock, more-itertools, moto, ncclient, packaging, pygit2, pynacl, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, vcert, websocket-client
 smmap2==2.0.5             # via gitdb2
 strict-rfc3339==0.7
 tempora==1.14.1           # via portend
diff --git a/requirements/static/py3.7/windows.txt b/requirements/static/py3.7/windows.txt
index 07cf52aeae92fd48ed3ba2c623c06e4c05c659cd..32ea9cedbb0db0d19252821d5dbaa9717a08a400 100644
--- a/requirements/static/py3.7/windows.txt
+++ b/requirements/static/py3.7/windows.txt
@@ -9,7 +9,6 @@ asn1crypto==0.24.0        # via cryptography
 atomicwrites==1.3.0       # via pytest
 attrs==19.1.0             # via pytest
 aws-xray-sdk==0.95        # via moto
-backports-abc==0.5
 backports.functools-lru-cache==1.5  # via cheroot
 backports.ssl-match-hostname==3.7.0.1
 boto3==1.9.132
@@ -101,8 +100,7 @@ s3transfer==0.2.0         # via boto3
 salttesting==2017.6.1
 sed==0.3.1
 setproctitle==1.1.10
-singledispatch==3.4.0.3
-six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, singledispatch, tempora, websocket-client
+six==1.12.0               # via cheroot, cherrypy, cryptography, docker, docker-pycreds, google-auth, kubernetes, mock, more-itertools, moto, packaging, pygit2, pyopenssl, pytest, python-dateutil, python-jose, pyvmomi, responses, salttesting, tempora, websocket-client
 smmap2==2.0.5             # via gitdb2
 smmap==0.9.0
 strict-rfc3339==0.7
diff --git a/setup.py b/setup.py
index 0ebe7ff8bd79542d3740d71e5d9e8e801b1d766c..06374647df5e82a21fc39b08d41c596f0483ff0c 100755
--- a/setup.py
+++ b/setup.py
@@ -206,7 +206,7 @@ def _parse_requirements_file(requirements_file):
             except ValueError:
                 pkg, pyverspec = line, ''
             pyverspec = pyverspec.strip()
-            if pyverspec:
+            if pyverspec and (not pkg.startswith('pycrypto') or pkg.startswith('pycryptodome')):
                 _, op, ver = pyverspec.split(' ', 2)
                 if not _check_ver(platform.python_version(), _parse_op(op), _parse_ver(ver)):
                     continue
-- 
2.23.0


